hnn_core/dipole.py

```diff
@@ -179,6 +173,10 @@ def read_dipole(fname):
         The instance of Dipole class
     """
 
+    # For supporting tests in test_gui.py
+    if isinstance(fname, StringIO):
+        return _read_dipole_txt(fname)
```

Comment:
you will want to copy the `read_dipole()` function from the master branch and replace it in your code as well